### PR TITLE
Implemented Sign Up and Sign In UI

### PR DESCRIPTION
**Pull request by Parthsimfor**
-There are two screens in UI ( Sign-Up and Sign-In)

- >Sign -In:
         - There are two TextFormField  and one Elevated button widgets are used.
- >Sign -Up:
         - There are four TextFormField and  one Elevated button widgets are used.

### CODE REVIEW

File: practical4/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# The following line activates a set of recommended lints for Flutter apps,
+# packages, and plugins designed to encourage good coding practices.
+include: package:flutter_lints/flutter.yaml
+
+linter:
+  # The lint rules applied to this project can be customized in the
+  # section below to disable rules from the `package:flutter_lints/flutter.yaml`
+  # included above or to enable additional rules. A list of all available lints
+  # and their documentation is published at
```
Comment (devarshranpara-simformsolutions) [inline practical4/analysis_options.yaml]: Remove comments

File: practical4/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      debugShowCheckedModeBanner: false,
+      theme: ThemeData(
+          inputDecorationTheme: InputDecorationTheme(
+        focusedBorder: UnderlineInputBorder(
+            borderSide: BorderSide(color: Colors.black, width: 2.5)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/main.dart]: Format missing

File: practical4/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      )),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/main.dart]: Format missing

File: practical4/lib/screens/Sign_in.dart [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      resizeToAvoidBottomInset: false,
+      body: Container(
+        width: double.infinity,
+        height: double.infinity,
+        // color: Colors.red,
+        decoration: const BoxDecoration(
+          image: DecorationImage(
+            fit: BoxFit.cover,
+            image: NetworkImage(
+                "https://images.pexels.com/photos/532168/pexels-photo-532168.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant here

File: practical4/lib/screens/Sign_in.dart [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+class _SignInState extends State<SignIn> {
+  final _logkey = GlobalKey<FormState>();
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      body: Container(
+        width: double.infinity,
+        height: double.infinity,
+        // color: Colors.red,
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Remove commented code.

File: practical4/lib/screens/Sign_in.dart [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        ),
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: const EdgeInsets.fromLTRB(25, 120, 0, 0),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                const Text(
+                  "Welcome",
+                  style: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use textStyle from the themedata

File: practical4/lib/screens/Sign_in.dart [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+                      color: Colors.green,
+                      fontWeight: FontWeight.bold,
+                      fontSize: 50,
+                      fontFamily: 'orbit'),
+                ),
+                const Text(
+                  "Login to access your account",
+                  style: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use textStyle from the themedata
Format missing

File: practical4/lib/screens/Sign_in.dart [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+                                  autofocus: true,
+                                  decoration: const InputDecoration(
+                                    labelText: "Email Address",
+                                    hintText: "Enter Username",
+                                    enabledBorder: UnderlineInputBorder(
+                                      borderSide:
+                                          BorderSide(color: Colors.black),
+                                    ),
+                                    labelStyle:
+                                        TextStyle(color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for colours.

File: practical4/lib/screens/Sign_in.dart [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+                                  ),
+                                  style: const TextStyle(
+                                    color: Colors.black,
+                                    fontWeight: FontWeight.bold,
+                                  ),
+                                  validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Exract this to a function
Add dataType to function

File: practical4/lib/screens/Sign_in.dart [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+                                      style:
+                                          TextStyle(color: Color(0xff0C3D27)),
+                                    ),
+                                    focusColor: Colors.black,
+                                    enabledBorder: UnderlineInputBorder(
+                                      borderSide:
+                                          BorderSide(color: Colors.black),
+                                    ),
+                                    labelStyle:
+                                        TextStyle(color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constants for colours.

File: practical4/lib/screens/Sign_in.dart [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+                                  ),
+                                  style: const TextStyle(
+                                    color: Colors.black,
+                                    fontWeight: FontWeight.bold,
+                                  ),
+                                  validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Extract this to function

File: practical4/lib/screens/Sign_in.dart [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+                                    if (value!.isEmpty) {
+                                      return "Please enter password";
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for Strings

File: practical4/lib/screens/Sign_in.dart [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+                              ),
+                              const SizedBox(
+                                height: 50,
+                              ),
+                              SizedBox(
+                                height: 50,
+                                width: 180,
+                                child: ElevatedButton(
+                                    onPressed: () {
+                                      FocusManager.instance.primaryFocus
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Extract this to function

File: practical4/lib/screens/Sign_in.dart [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+                                      }
+                                    },
+                                    style: ElevatedButton.styleFrom(
+                                        backgroundColor:
+                                            const Color(0xff0C3D27),
+                                        shape: RoundedRectangleBorder(
+                                            borderRadius:
+                                                BorderRadius.circular(50),
+                                            side: const BorderSide(
+                                                color: Color(0xff0C3D27)))),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for colours
Format missing

File: practical4/lib/screens/Sign_in.dart [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+                                    child: const Text('Sign In')),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for theme

File: practical4/lib/screens/Sign_in.dart [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+                                width: 180,
+                                child: ElevatedButton(
+                                    onPressed: () {},
+                                    style: ElevatedButton.styleFrom(
+                                        backgroundColor: Colors.white,
+                                        shape: RoundedRectangleBorder(
+                                            borderRadius:
+                                                BorderRadius.circular(50),
+                                            side: const BorderSide(
+                                              color: Color(0xff0C3D27),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for colour

File: practical4/lib/screens/Sign_in.dart [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+                                            ))),
+                                    child: const Text(
+                                      'Sign in with Google',
+                                      style:
+                                          TextStyle(color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for colour

File: practical4/lib/screens/Sign_in.dart [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+                                    style: ElevatedButton.styleFrom(
+                                        backgroundColor: Colors.white,
+                                        shape: RoundedRectangleBorder(
+                                            borderRadius:
+                                                BorderRadius.circular(50),
+                                            side: const BorderSide(
+                                              color: Color(0xff0C3D27),
+                                            ))),
+                                    child: const Text(
+                                      'Sign in with Google',
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Use constant for String

File: practical4/lib/screens/Sign_in.dart [hunk 19]
```diff
@@ -0,0 +1,235 @@
...
+                    children: [
+                      const Text(
+                        ("Don't have an account? "),
+                        style: TextStyle(
+                            color: Colors.white,
+                            fontSize: 15,
+                            fontWeight: FontWeight.bold),
+                      ),
+                      GestureDetector(
+                        onTap: () {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Extract this to function

File: practical4/lib/screens/sign_up.dart [hunk 20]
```diff
@@ -0,0 +1,264 @@
...
+        resizeToAvoidBottomInset: false,
+        body: Container(
+          width: double.infinity,
+          height: double.infinity,
+          // color: Colors.red,
+          decoration: const BoxDecoration(
+            image: DecorationImage(
+              fit: BoxFit.cover,
+              image: NetworkImage(
+                  "https://images.pexels.com/photos/532168/pexels-photo-532168.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use same constant here as well for network image

File: practical4/lib/screens/sign_up.dart [hunk 21]
```diff
@@ -0,0 +1,264 @@
...
+  final _formKey = GlobalKey<FormState>();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        resizeToAvoidBottomInset: false,
+        body: Container(
+          width: double.infinity,
+          height: double.infinity,
+          // color: Colors.red,
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Remove commented code

File: practical4/lib/screens/sign_up.dart [hunk 22]
```diff
@@ -0,0 +1,264 @@
...
+          child: SingleChildScrollView(
+            child: Container(
+              child: Padding(
+                padding: const EdgeInsets.fromLTRB(25, 120, 0, 0),
+                child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      const Text(
+                        "Register",
+                        style: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use text from themedata

File: practical4/lib/screens/sign_up.dart [hunk 23]
```diff
@@ -0,0 +1,264 @@
...
+                            color: Colors.green,
+                            fontWeight: FontWeight.w500,
+                            fontSize: 50,
+                            fontFamily: 'orbit'),
+                      ),
+                      const Text(
+                        "Create a new account !!",
+                        style: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use text from themedata

File: practical4/lib/screens/sign_up.dart [hunk 24]
```diff
@@ -0,0 +1,264 @@
...
+                                        autofocus: true,
+                                        decoration: const InputDecoration(
+                                          labelText: "Full name",
+                                          hintText: "Enter full name",
+                                          enabledBorder: UnderlineInputBorder(
+                                            borderSide:
+                                                BorderSide(color: Colors.black),
+                                          ),
+                                          labelStyle: TextStyle(
+                                              color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use text from themedata

File: practical4/lib/screens/sign_up.dart [hunk 25]
```diff
@@ -0,0 +1,264 @@
...
+                                        ),
+                                        style: const TextStyle(
+                                          color: Colors.black,
+                                          fontWeight: FontWeight.bold,
+                                        ),
+                                        validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to function

File: practical4/lib/screens/sign_up.dart [hunk 26]
```diff
@@ -0,0 +1,264 @@
...
+                                      ),
+                                    ),
+                                    const SizedBox(
+                                      height: 30,
+                                    ),
+                                    SizedBox(
+                                      width: 290,
+                                      child: TextFormField(
+                                        decoration: const InputDecoration(
+                                          labelText: "Your email address",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constants for Strings

File: practical4/lib/screens/sign_up.dart [hunk 27]
```diff
@@ -0,0 +1,264 @@
...
+                                          hintText: "Enter email id",
+                                          focusColor: Colors.black,
+                                          enabledBorder: UnderlineInputBorder(
+                                            borderSide:
+                                                BorderSide(color: Colors.black),
+                                          ),
+                                          labelStyle: TextStyle(
+                                              color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant here.

File: practical4/lib/screens/sign_up.dart [hunk 28]
```diff
@@ -0,0 +1,264 @@
...
+                                        ),
+                                        style: const TextStyle(
+                                          color: Colors.black,
+                                          fontWeight: FontWeight.bold,
+                                        ),
+                                        validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to funcntion.
Define datatype here

File: practical4/lib/screens/sign_up.dart [hunk 29]
```diff
@@ -0,0 +1,264 @@
...
+                                    ),
+                                    const SizedBox(
+                                      height: 30,
+                                    ),
+                                    SizedBox(
+                                      width: 290,
+                                      child: TextFormField(
+                                        obscureText: true,
+                                        decoration: const InputDecoration(
+                                          labelText: "Password",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant here.

File: practical4/lib/screens/sign_up.dart [hunk 30]
```diff
@@ -0,0 +1,264 @@
...
+                                          hintText: "Enter your new password",
+                                          focusColor: Colors.black,
+                                          enabledBorder: UnderlineInputBorder(
+                                            borderSide:
+                                                BorderSide(color: Colors.black),
+                                          ),
+                                          labelStyle: TextStyle(
+                                              color: Color(0xff0C3D27)),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant here.

File: practical4/lib/screens/sign_up.dart [hunk 31]
```diff
@@ -0,0 +1,264 @@
...
+                                        ),
+                                        style: const TextStyle(
+                                          color: Colors.black,
+                                          fontWeight: FontWeight.bold,
+                                        ),
+                                        validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to function

File: practical4/lib/screens/sign_up.dart [hunk 32]
```diff
@@ -0,0 +1,264 @@
...
+                                    ),
+                                    const SizedBox(
+                                      height: 30,
+                                    ),
+                                    SizedBox(
+                                      width: 290,
+                                      child: TextFormField(
+                                        obscureText: true,
+                                        decoration: const InputDecoration(
+                                          labelText: "Confirm Password",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant here.

File: practical4/lib/screens/sign_up.dart [hunk 33]
```diff
@@ -0,0 +1,264 @@
...
+                                                BorderSide(color: Colors.black),
+                                          ),
+                                          labelStyle: TextStyle(
+                                              color: Color(0xff0C3D27)),
+                                        ),
+                                        style: const TextStyle(
+                                          color: Colors.black,
+                                          fontWeight: FontWeight.bold,
+                                        ),
+                                        validator: (value) {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to function

File: practical4/lib/screens/sign_up.dart [hunk 34]
```diff
@@ -0,0 +1,264 @@
...
+                                      ),
+                                    ),
+                                    const SizedBox(
+                                      height: 30,
+                                    ),
+                                    Container(
+                                        height: 50,
+                                        width: 180,
+                                        child: ElevatedButton(
+                                            onPressed: () {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to function

File: practical4/lib/screens/sign_up.dart [hunk 35]
```diff
@@ -0,0 +1,264 @@
...
+                          children: [
+                            const Text(
+                              ("Already have account? "),
+                              style: TextStyle(
+                                  color: Colors.white,
+                                  fontSize: 15,
+                                  fontWeight: FontWeight.bold),
+                            ),
+                            GestureDetector(
+                              onTap: () {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Extract this to function

File: practical4/pubspec.yaml [hunk 36]
```diff
@@ -0,0 +1,90 @@
+name: practical4
+description: A new Flutter project.
+# The following line prevents the package from being accidentally published to
+# pub.dev using `flutter pub publish`. This is preferred for private packages.
+publish_to: 'none' # Remove this line if you wish to publish to pub.dev
+
+# The following defines the version and build number for your application.
+# A version number is three numbers separated by dots, like 1.2.43
+# followed by an optional build number separated by a +.
+# Both the version and the builder number may be overridden in flutter
```
Comment (devarshranpara-simformsolutions) [inline practical4/pubspec.yaml]: Remove commented code

File: practical4/pubspec.yaml [hunk 37]
```diff
@@ -0,0 +1,90 @@
...
+# Read more about iOS versioning at
+# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
+# In Windows, build-name is used as the major, minor, and patch parts
+# of the product and file versions while build-number is used as the build suffix.
+version: 1.0.0+1
+
+environment:
+  sdk: '>=2.19.0 <3.0.0'
+
+# Dependencies specify other packages that your package needs in order to work.
```
Comment (devarshranpara-simformsolutions) [inline practical4/pubspec.yaml]: Remove comments

**Review by devarshranpara-simformsolutions — CHANGES_REQUESTED** [review]

File: practical4/pubspec.yaml (hunk 37 above)
Comment (Parthsimfor) [reply practical4/pubspec.yaml]: Resolved

File: practical4/analysis_options.yaml (hunk 1 above)
Comment (Parthsimfor) [reply practical4/analysis_options.yaml]: Resolved

File: practical4/lib/main.dart [hunk 38]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      debugShowCheckedModeBanner: false,
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/main.dart]: Please remove this flag

File: practical4/lib/main.dart [hunk 39]
```diff
@@ -0,0 +1,45 @@
...
+        inputDecorationTheme: const InputDecorationTheme(
+          focusedBorder: UnderlineInputBorder(
+            borderSide: BorderSide(
+              color: Colors.black,
+              width: 2.5,
+            ),
+          ),
+        ),
+      ),
+      home: const SignUp(),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/main.dart]: Please use named routes.

File: practical4/lib/screens/Sign_in.dart [hunk 40]
```diff
@@ -0,0 +1,269 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SingleChildScrollView(
+        child: Container(
+          height: MediaQuery.of(context).size.height,
+          decoration: const BoxDecoration(
+            image: const DecorationImage(
+              fit: BoxFit.cover,
+              image: const AssetImage('assets/image/MainImage.png'),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use constat here.

File: practical4/lib/screens/Sign_in.dart [hunk 41]
```diff
@@ -0,0 +1,269 @@
...
+                                height: 40,
+                              ),
+                              SizedBox(
+                                width: 280,
+                                child: TextFormField(
+                                  obscureText: true,
+                                  textAlignVertical: TextAlignVertical.bottom,
+                                  decoration: InputDecoration(
+                                    hintText: "Enter Password",
+                                    //labelText: "Enter Password",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please remove commented code.

File: practical4/lib/screens/Sign_in.dart [hunk 42]
```diff
@@ -0,0 +1,269 @@
...
+                                    suffixIcon: GestureDetector(
+                                      onTap: () {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please extract this to function

File: practical4/lib/screens/Sign_in.dart [hunk 43]
```diff
@@ -0,0 +1,269 @@
...
+                                        Navigator.push(
+                                          context,
+                                          MaterialPageRoute(
+                                            builder: (context) =>
+                                                const Forget(),
+                                          ),
+                                        );
+                                      },
+                                      child: const Text(
+                                        "Forgot ?",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use constat here for String.

File: practical4/lib/screens/Sign_in.dart [hunk 44]
```diff
@@ -0,0 +1,269 @@
...
+                                        style: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use textStyle from themedata

File: practical4/lib/screens/Sign_in.dart [hunk 45]
```diff
@@ -0,0 +1,269 @@
...
+                                          color: CommonColors.greencolor,
+                                        ),
+                                      ),
+                                    ),
+                                    focusColor: CommonColors.blackColor,
+                                    enabledBorder: UnderlineInputBorder(
+                                      borderSide: BorderSide(
+                                          color: CommonColors.blackColor),
+                                    ),
+                                    labelStyle: TextStyle(
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use textStyle from themedata

File: practical4/lib/screens/Sign_in.dart [hunk 46]
```diff
@@ -0,0 +1,269 @@
...
+                                  style: ElevatedButton.styleFrom(
+                                    backgroundColor: CommonColors.greencolor,
+                                    shape: RoundedRectangleBorder(
+                                      borderRadius: BorderRadius.circular(50),
+                                      side: const BorderSide(
+                                        color: CommonColors.greencolor,
+                                      ),
+                                    ),
+                                  ),
+                                  child: const Text('Sign In'),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use Strting constant

File: practical4/lib/screens/Sign_in.dart [hunk 47]
```diff
@@ -0,0 +1,269 @@
...
+                                child: ElevatedButton(
+                                  onPressed: () {},
+                                  style: ElevatedButton.styleFrom(
+                                      backgroundColor: CommonColors.whiteColor,
+                                      shape: RoundedRectangleBorder(
+                                          borderRadius:
+                                              BorderRadius.circular(50),
+                                          side: const BorderSide(
+                                            color: CommonColors.greencolor,
+                                          ))),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Format missing

File: practical4/lib/screens/Sign_in.dart [hunk 48]
```diff
@@ -0,0 +1,269 @@
...
+                ),
+                const SizedBox(
+                  height: 50,
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(75, 0, 0, 0),
+                  child: Row(
+                    children: [
+                      const Text(
+                        ("Don't have an account? "),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/Sign_in.dart]: Please use constant, remove breakets

File: practical4/lib/screens/Sign_in.dart (hunk 19 above)
Comment (devarshranpara-simformsolutions) [reply practical4/lib/screens/Sign_in.dart]: This is not resolved!
Please fix this comment

File: practical4/lib/screens/forget.dart [hunk 49]
```diff
@@ -0,0 +1,155 @@
...
+
+  bool isEmailValid(String value) => RegExp(Strings.emailRegEx).hasMatch(value);
+
+  void conditionToShowAlertDialog() {
+    if (_logkey.currentState!.validate()) {
+      forgetPassword();
+    }
+  }
+
+  forgetPassword() {
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/forget.dart]: Please add return type

File: practical4/lib/screens/forget.dart [hunk 50]
```diff
@@ -0,0 +1,155 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      body: Container(
+        width: double.infinity,
+        height: double.infinity,
+        decoration: const BoxDecoration(
+          image: DecorationImage(
+            fit: BoxFit.cover,
+            image: AssetImage('assets/image/MainImage.png'),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/forget.dart]: Please use contant here

File: practical4/lib/screens/forget.dart [hunk 51]
```diff
@@ -0,0 +1,155 @@
...
+                            children: [
+                              SizedBox(
+                                width: 280,
+                                child: TextFormField(
+                                  keyboardType: TextInputType.emailAddress,
+                                  autovalidateMode:
+                                      AutovalidateMode.onUserInteraction,
+                                  autofocus: true,
+                                  decoration: const InputDecoration(
+                                    labelText: "Email Address",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/forget.dart]: Please use constant here

File: practical4/lib/screens/forget.dart [hunk 52]
```diff
@@ -0,0 +1,155 @@
...
+                                  onPressed: () => conditionToShowAlertDialog(),
+                                  style: ElevatedButton.styleFrom(
+                                    backgroundColor: CommonColors.greencolor,
+                                    shape: RoundedRectangleBorder(
+                                      borderRadius: BorderRadius.circular(50),
+                                      side: const BorderSide(
+                                          color: CommonColors.greencolor),
+                                    ),
+                                  ),
+                                  child: const Text('Forget Password'),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/forget.dart]: Please use constent here.

File: practical4/lib/screens/sign_up.dart [hunk 53]
```diff
@@ -0,0 +1,276 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      body: Container(
+        width: double.infinity,
+        height: double.infinity,
+        decoration: const BoxDecoration(
+          image: DecorationImage(
+            fit: BoxFit.cover,
+            image: AssetImage('assets/image/MainImage.png'),
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant

File: practical4/lib/screens/sign_up.dart [hunk 54]
```diff
@@ -0,0 +1,276 @@
...
+        child: SingleChildScrollView(
+          child: SizedBox(
+            height: 950,
+            child: Padding(
+              padding: const EdgeInsets.fromLTRB(25, 120, 0, 0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    "Register",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant

File: practical4/lib/screens/sign_up.dart [hunk 55]
```diff
@@ -0,0 +1,276 @@
...
+                    style: Theme.of(context).textTheme.displayLarge,
+                  ),
+                  Text(
+                    "Create a new account !!",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant

File: practical4/lib/screens/sign_up.dart [hunk 56]
```diff
@@ -0,0 +1,276 @@
...
+                    style: Theme.of(context).textTheme.displayMedium,
+                  ),
+                  const SizedBox(
+                    height: 20,
+                  ),
+                  Stack(
+                    children: [
+                      Container(
+                        //height: 455,
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Remove commented code.

File: practical4/lib/screens/sign_up.dart [hunk 57]
```diff
@@ -0,0 +1,276 @@
...
+                                    ),
+                                  ),
+                                  const SizedBox(
+                                    height: 30,
+                                  ),
+                                  SizedBox(
+                                    width: 290,
+                                    child: TextFormField(
+                                      decoration: const InputDecoration(
+                                        hintText: "Enter email id",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Remove commented code.

File: practical4/lib/screens/sign_up.dart [hunk 58]
```diff
@@ -0,0 +1,276 @@
...
+                                  const SizedBox(
+                                    height: 30,
+                                  ),
+                                  SizedBox(
+                                    width: 290,
+                                    child: TextFormField(
+                                      controller: password,
+                                      obscureText: true,
+                                      decoration: const InputDecoration(
+                                        hintText: "Enter password",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant

File: practical4/lib/screens/sign_up.dart [hunk 59]
```diff
@@ -0,0 +1,276 @@
...
+                                  const SizedBox(
+                                    height: 30,
+                                  ),
+                                  SizedBox(
+                                    width: 290,
+                                    child: TextFormField(
+                                      controller: confirmPassword,
+                                      obscureText: true,
+                                      decoration: const InputDecoration(
+                                        hintText: "Confirm password",
```
Comment (devarshranpara-simformsolutions) [inline practical4/lib/screens/sign_up.dart]: Use constant

**Review by devarshranpara-simformsolutions — CHANGES_REQUESTED** [review]

File: practical4/lib/screens/Sign_in.dart (hunk 42 above)
Comment (devarshranpara-simformsolutions) [reply practical4/lib/screens/Sign_in.dart]: This comment is pending

**Review by devarshranpara-simformsolutions — CHANGES_REQUESTED** [review]

**Review by devarshranpara-simformsolutions — APPROVED** [review]

**Pull request merged**